### PR TITLE
Feature photos simplify checks

### PR DESCRIPTION
**Pull request by Fasse**
- use methods for checks of visible and editable of photo albums
- move lock and unlock mode to photo_album_function

### CODE REVIEW

File: adm_program/modules/photos/photo_album_function.php [hunk 1]
```diff
@@ -30,39 +30,29 @@
...
+    $gMessage->show($gL10n->get('PHO_NO_RIGHTS'));
+    // => EXIT
 }
 
 $phoId = (int) $photoAlbum->getValue('pho_id');
 
 // Speicherort mit dem Pfad aus der Datenbank
 $albumPath = ADMIDIO_PATH . FOLDER_DATA . '/photos/' . $photoAlbum->getValue('pho_begin', 'Y-m-d') . '_' . $phoId;
 
-$phoBegin = $_POST['pho_begin'];
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/modules/photos/photo_album_function.php]: you convert this to local copy but the logic was broken. This is a quick fix to let it work as before!

File: adm_program/modules/photos/photo_album_function.php [hunk 2]
```diff
@@ -81,17 +71,17 @@
...
+        $startDate = \DateTime::createFromFormat($gPreferences['system_date'], $_POST['pho_begin']);
         if ($startDate === false)
         {
             $gMessage->show($gL10n->get('SYS_DATE_INVALID', array($gL10n->get('SYS_START'), $gPreferences['system_date'])));
             // => EXIT
         }
         else
         {
-            $phoBegin = $startDate->format('Y-m-d');
+            $_POST['pho_begin'] = $startDate->format('Y-m-d');
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/modules/photos/photo_album_function.php]: you convert this to local copy but the logic was broken. This is a quick fix to let it work as before!

File: adm_program/modules/photos/photo_album_function.php [hunk 3]
```diff
@@ -101,26 +91,26 @@
...
+        $endDate = \DateTime::createFromFormat($gPreferences['system_date'], $_POST['pho_end']);
         if ($endDate === false)
         {
             $gMessage->show($gL10n->get('SYS_DATE_INVALID', array($gL10n->get('SYS_END'), $gPreferences['system_date'])));
             // => EXIT
         }
         else
         {
-            $phoEnd = $endDate->format('Y-m-d');
+            $_POST['pho_end'] = $endDate->format('Y-m-d');
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/modules/photos/photo_album_function.php]: you convert this to local copy but the logic was broken. This is a quick fix to let it work as before!

File: adm_program/modules/photos/photo_album_function.php [hunk 4]
```diff
@@ -101,26 +91,26 @@
...
         }
     }
     else
     {
-        $phoEnd = $phoBegin;
+        $_POST['pho_end'] = $_POST['pho_begin'];
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/modules/photos/photo_album_function.php]: you convert this to local copy but the logic was broken. This is a quick fix to let it work as before!

File: adm_program/modules/photos/photo_function.php [hunk 5]
```diff
@@ -88,13 +88,8 @@ function tryRename($path, $newPath)
  * @param int $phoId
  * @param int $picNr
  */
-function deletePhoto($phoId, $picNr)
+function deletePhoto(TablePhotos $photoAlbum, $phoId, $picNr)
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/modules/photos/photo_function.php]: done

File: adm_program/system/classes/tablephotos.php [hunk 6]
```diff
@@ -298,14 +298,14 @@ public function visible()
         global $gCurrentOrganization, $gCurrentUser;
 
         // current photo album must belong to current organization
-        if($this->getValue('pho_id') > 0 && (int) $this->getValue('pho_org_id') === (int) $gCurrentOrganization->getValue('org_id'))
+        if($this->getValue('pho_id') > 0 && (int) $this->getValue('pho_org_id') !== (int) $gCurrentOrganization->getValue('org_id'))
         {
-            return true;
+            return false;
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/system/classes/tablephotos.php]: done

File: adm_program/system/classes/tablephotos.php [hunk 7]
```diff
@@ -298,14 +298,14 @@ public function visible()
...
         }
         // locked photo album could only be viewed by module administrators
-        elseif((int) $this->getValue('pho_locked') === 0 || ((int) $this->getValue('pho_locked') === 1 && $gCurrentUser->editPhotoRight()))
+        elseif((int) $this->getValue('pho_locked') === 1 && !$gCurrentUser->editPhotoRight())
         {
-            return true;
+            return false;
```
[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/system/classes/tablephotos.php]: done

**Comment by Fasse** [comment]
@ximex please have a look at my comments.

File: adm_program/modules/photos/photo_album_function.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ximex) [reply adm_program/modules/photos/photo_album_function.php]: Ok than we will look later at this.

**Pull request merged**